doc/tutorial/ranges/range_solution4.cpp

```diff
@@ -0,0 +1,36 @@
+#include <vector>
+
+#include <seqan3/alphabet/nucleotide/dna4.hpp>
+#include <seqan3/argument_parser/all.hpp>                       // include argument parser
+#include <seqan3/range/container/bitcompressed_vector.hpp>      // include bitcompressed vector
+
+using namespace seqan3;
+
+int main(int argc, char ** argv)
+{
+    argument_parser myparser("Vector implementations comparison", argc, argv);
+    size_t size{};
+    bool use_bitvector{};
+    myparser.add_positional_option(size, "Size of vector");
+    myparser.add_flag(use_bitvector, 'b', "bitvector", "Use bitvector instead of vector");
+    try
+    {
+         myparser.parse();
+    }
+    catch (parser_invalid_argument const & ext)                     // catch user errors
+    {
+        debug_stream << "[Error] " << ext.what() << "\n";
+        return -1;
+    }
+    
+    if (use_bitvector)
+    {
+        bitcompressed_vector<dna4> vector;
+        vector.resize(size, 'A'_dna4);
+    }
+    else
+    {
+        std::vector<dna4> vector;
```

Comment:
you can directly initialise the vector with size `size`:

 ```suggestion
        std::vector<dna4> vector{size};
```
no resizing needed. Check if this also works for our bitcompressed_vector. 

Also I am not sure if this might be optimized away, so maybe you want to print the size to prevent this
```cpp
debug_stream << "Allocated std::vector<dna4> of size " << vector.size() << std::endl;
```
Maybe drop a note in the exercise description that one should print the size.

Reply:
Okay, I changed the initialization of the std::vector but this is not possible with bitcompressed_vector.